tests/suites/hup/tests/storagemigration.js

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2021 balena
+ *
+ * @license Apache-2.0
+ */
+
+'use strict';
+
+const shouldMigrate = async (that, test) => {
+  // check if both aufs and overlayfs are supported, otherwise skip
+  if (await that.context.get().worker.executeCommandInHostOS(
+      `balena info 2>/dev/null | grep -o -e aufs -e overlay2`,
+      that.context.get().link,
+    ) === "overlay2") {
+    test.comment(`SKIP: Already using overlayfs, skipping migration test.`)
+    return false;
+  }
+  if (await that.context.get().worker.executeCommandInHostOS(
+      `grep -e overlay -e aufs /proc/filesystems >/dev/null && echo pass`,
+      that.context.get().link,
+    ) !== "pass") {
+    test.comment(`SKIP: Both aufs and overlayfs have to be supported to run a migration.`)
+    return false;
+  }
+  return true;
+}
+
+const archiveMigrationLogs = async (that, test) => {
+  test.comment(`Archiving migration logs`);
+  const migration = await that.context.get().worker.executeCommandInHostOS(
+      `set -x; journalctl -u balena -t balenad | grep 'to overlay2' || true`,
+      that.context.get().link);
+  const migrationLogs = require("path").join(that.suite.options.tmpdir, `storage-migration.log`);
+  require("fs").writeFileSync(migrationLogs, migration);
+  await that.archiver.add(migrationLogs);
+  require("fs").unlinkSync(migrationLogs);
+}
+
+const waitForRollbackHealth = async (that, test) => {
+  test.comment(`Waiting for rollback-health...`);
+  await that.context.get().utils.waitUntil(async () => {
+    return (
+      (await that.context.get().worker.executeCommandInHostOS(
+        `systemctl status rollback-health.service`,
+        that.context.get().link,
+      )) !== 'active'
```

Comment:
Take a look at the recently merged updates to the smoke test in https://github.com/balena-os/meta-balena/pull/2238

`systemctl status` will never return `active`, do you mean `systemctl is-active`?

Also the rollback-health service will be active for a minute or two before rebooting again if there is a problem, so you might also want to make sure the breadcrumbs are gone.